packages/optimizer/spec/transformers/valid/ReorderHeadTransformer/supports_esm/expected_output.html

```diff
@@ -0,0 +1,16 @@
+<!doctype html>
+<html ⚡>
+<head>
+  <link rel="stylesheet" href="https://cdn.ampproject.org/v0.css">
+  <script async src="https://cdn.ampproject.org/v0.mjs" type="module" crossorigin="anonymous"></script>
+  <script async nomodule src="https://cdn.ampproject.org/v0/amp-experiment-0.1.js" custom-element="amp-experiment"></script>
+  <script async custom-element="amp-experiment" src="https://cdn.ampproject.org/v0/amp-experiment-0.1.mjs" type="module" crossorigin="anonymous"></script>
+  <script async nomodule src="https://cdn.ampproject.org/v0/amp-mustache-0.2.js" custom-template="amp-mustache"></script>
+  <script async custom-template="amp-mustache" src="https://cdn.ampproject.org/v0/amp-mustache-0.2.mjs" type="module" crossorigin="anonymous"></script>
+  <link href="https://example.com/favicon.ico" rel="icon">
+  <link rel="preload" href="https://cdn.ampproject.org/v0.css" as="style">
+  <link href="https://fonts.googleapis.com/css?foobar" rel="stylesheet" type="text/css">
+  <link as="script" crossorigin="anonymous" href="https://cdn.ampproject.org/v0.mjs" rel="modulepreload">
```

Comment:
Shouldn't the `link[rel=preload]` and `link[rel=modulepreload]` be moved to the beginning of the `head`?

Reply:
Done